risc0/zkvm/src/exec/env.rs

```diff
@@ -151,6 +213,39 @@ impl<'a> ExecutorEnvBuilder<'a> {
     }
 
     /// Add initial input that can be read by the guest from stdin.
+    /// This step must occur before the executor environment is built.
+    /// # Examples
+    /// ```
+    /// # use risc0_zkvm::{
+    /// #   ExecutorEnv,
+    /// #   ExecutorEnvBuilder,
+    ///     serde::to_vec};
+    ///
+    /// let a: u64 = 400;
+    ///
+    /// let env = ExecutorEnv::builder()
+    ///     .add_input(&to_vec(&a).unwrap())
+    ///     .build();
+    /// ```
+    ///
+    /// Most of the time, the best practice is to combine guest program inputs
+    /// into a single struct. For cases where you want to add inputs that
+    /// the guest will access using consecutive reads, you can call
+    /// `ExecutorEnvBuilder::add_input()` iteratively:
```

Comment:
Instead of making a prescription, I'd just say what the behavior of calling `add_input` is, which is to concatenate the inputs together

Reply:
Removing the prescriptive first line but making note that guest can access concatenated input with consecutive reads, can take it out if that's still too prescriptive.